vllm/v1/worker/gpu_model_runner.py

```diff
@@ -3871,17 +3893,44 @@ def _capture_cudagraphs(
                 )
             )
 
+            # Force attention for all cudagraph modes when the backend forward
+            # op doesn't include KV cache update. This is required
+            # for KV cache update to be captured correctly in cases where
+            # the KV cache update and attention are two separate custom ops.
+            #
+            # Use CUDAGraphRuntimeStyle.NONE (default) for warmup.
+            # But be careful, warm up with `NONE` is orthogonal to
+            # if we want to warm up attention or not. This is
+            # different from the case where `FULL` implies capture all
+            # attention while `PIECEWISE` implies no attention for backends
+            # that don't split KV Cache update and attention op.
+            has_separate_kv_update = True  # not all(\
```

Comment:
Comment? Also can you merge the two paragraphs above into a coherent comment about when attention is enabled or not